Jenkinsfile

```diff
@@ -38,6 +44,23 @@ pipeline {
             }
         }
 
+        stage('Prepare-Logs-Destination') {
+            agent { label 'master' }
+            steps {
+                script {
+                    if (env.ZUUL_UUID != null) {
+                        logsRemoteDir = "${logRootDir}/${env.ZUUL_UUID}"
```

Comment:
See, this is weird. If we were to go with the amazing `pushToLogServer(filename, relative_filepath_on_the_fileserver)`, then we need to tell it (inject) the `logsRemoteDir`. Which makes me think that it's actually a higher order function. 

We can basically construct our pushToLogServer function by injecting rootDir like this:
```
rootDir = "asdf/asdf/asdf/"
def pushLogs = { subdir -> rootDir + subdir }
print pushLogs ("myPrivateDir/asdf.txt")

// prints asdf/asdf/asdf/myPrivateDir/asdf.txt
```
Then, rootDir doesn't have to be global. However, pushLogs function would be global. But I think it's much better.

Reply:
sry it should be something like
```
def pushLogs
if(true) { // simulate block
    rootDir = "asdf/asdf/asdf/"
    pushLogs = { subdir -> rootDir + subdir }
}
print pushLogs ("myPrivateDir/asdf.txt")
```